osu.Game.Tournament/Screens/Gameplay/GameplayScreen.cs

```diff
@@ -152,6 +181,35 @@ protected override void CurrentMatchChanged(ValueChangedEvent<TournamentMatch> m
             scheduledOperation?.Cancel();
         }
 
+        private void beatmapChanged(ValueChangedEvent<BeatmapInfo> beatmap)
+        {
+            if (beatmap.NewValue == null || CurrentMatch.Value == null)
+                return;
+
+            var currentSelectedBeatmapMods = CurrentMatch.Value.Round.Value.Beatmaps.FirstOrDefault(b => b.ID == beatmap.NewValue.OnlineBeatmapID)?.Mods;
+            showVideoForModsAndHideAllElse(currentSelectedBeatmapMods);
+        }
+
+        private void showVideoForModsAndHideAllElse(string currentSelectedBeatmapMods)
+        {
+            float fallbackVidAlpha = 1f;
+
+            foreach (var (mods, video) in modSpecificGameplayVideos)
+            {
+                if (mods.Equals(currentSelectedBeatmapMods, StringComparison.OrdinalIgnoreCase))
+                {
+                    video.Alpha = 1;
+                    fallbackVidAlpha = 0;
```

Comment:
sure you don't want any fade transitions here? having the background change instantly like that wouldn't look very good on a broadcast i think

Reply:
My thinking was that beatmap changes happen mostly on the mappool scene so you probably wouldnt really see the change, but I guess fair enough. I'll be honest im not experianced enough with the osu framework to know how to implement a fade transition here.

also now that I type this, maybe this should then also be implemented on the mappool screen? didnt really think about that tbh

Reply:
I made an attempt on implementing a fade transition, lemmy know if I did it correctly.

I dont know if the `FadeIn` and `FadeOut` is smart enough to do nothing if its already in the desired state? Or do I need to add that check as well to not do unnecessary work?

Reply:
As for the question above, fades are no-ops if the item already has the correct alpha (it is read into the transform at the time of application), yes.